### PR TITLE
[TACHYON-1155] Add toString method for tachyon.client.file.options pa…

### PR DESCRIPTION
**Pull request by miasantreble**
…ckage

https://tachyon.atlassian.net/browse/TACHYON-1155

Some tests (like `FileOutStreamIntegrationTest` use `options.toString()` as temporary file names which is why no space is used in the override `toString()` impl.

### CODE REVIEW

**Comment by yupeng9** [comment]
Jenkins, test this please

File: clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java [hunk 1]
```diff
@@ -143,6 +143,15 @@ public UnderStorageType getUnderStorageType() {
   }
 
   /**
+   * @return the name : value pairs for all the fields
+   */
+  @Override
+  public String toString() {
+    return "CreateTOptions{mBlockSizeBytes=" + mBlockSizeBytes + ",mRecursive=" + mRecursive
```
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: How about using Guava's toStringHelper for this?
http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/base/MoreObjects.html#toStringHelper(java.lang.Object)

Comment (miasantreble) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: looks like toStringHelper will insert a space between fields (like  
```
// Returns "ClassName{x=1, y=foo}"
```
), which will be a problem unless we rewrite the tests

Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: @miasantreble I think having spaces required to make the output human friendly. How about making another method for the tests?

Comment (miasantreble) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: @calvinjia Agreed on readability with addition of spaces, I can make a helper function toStringBase() to call super.toString() specifically for tests, plus replace string literals with toStringHelper. How does that sound?

Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: @miasantreble I don't think its necessary to upgrade Guava just for this method.

Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: Or you can strip the spaces for test. Also, if we do not upgrade guava, what about we add a similar util method, which will be handy for all toString methods. And we can easily replace it when we feel it's time to upgrade Guava.

**Comment by miasantreble** [comment]
jenkins, test this please.

**Comment by miasantreble** [comment]
jenkins, test this please

**Comment by jsimsa** [comment]
@miasantreble thank you for the PR. Could you please explain what is this change for?

**Comment by miasantreble** [comment]
@calvinjia I assume this is to help with logging and debugging?

**Comment by calvinjia** [comment]
@jsimsa @miasantreble 
This change would mostly be helpful for logging by making it easy to get a human readable representation of an option.

File: clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java [hunk 2]
```diff
@@ -143,6 +143,22 @@ public UnderStorageType getUnderStorageType() {
   }
 
   /**
+   * @return the name : value pairs for all the fields
+   */
+  @Override
+  public String toString() {
+    return "CreateOptions{mBlockSizeBytes=" + mBlockSizeBytes + ", mRecursive=" + mRecursive
+        + ", mTTL=" + mTTL + ", mUnderStorageType=" + mUnderStorageType.toString() + "}";
```
Comment (calvinjia) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: Could you follow the object toString format in classes like `InodeFile`?

File: clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java [hunk 3]
```diff
@@ -143,6 +143,22 @@ public UnderStorageType getUnderStorageType() {
...
+  }
+
+  /**
+   * @return the default String representation for testing purposes
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: I think it is unnecessary to add these methods for the sake of unit tests.

If the tests want a string representation of the object with no spaces, they can do: `op.toString().st.replaceAll("\\s+","")` and the tests don't care about the string being meaninful, they can use `op.hashCode()`

Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: +1, it's better to do the string processing within the tests.

**Comment by calvinjia** [comment]
LGTM

File: clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java [hunk 4]
```diff
@@ -143,6 +143,20 @@ public UnderStorageType getUnderStorageType() {
   }
 
   /**
+   * @return the name : value pairs for all the fields
+   */
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder("CreateOptions(");
+    sb.append(super.toString()).append(", mBlockSizeBytes: ").append(mBlockSizeBytes);
```
Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: Remove the leading `m` from all the descriptions. This is used by the implementation to distinguish member variables, but there is no need to propagate them to logs.

Comment (jsimsa) [inline clients/unshaded/src/main/java/tachyon/client/file/options/CreateOptions.java]: This applies to all of the `toString()` methods.

**Comment by jsimsa** [comment]
One minor comments. Other than that it looks good to me.

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**